cluster/plan.go

```diff
@@ -512,7 +512,8 @@ func (c *Cluster) BuildKubeletProcess(host *hosts.Host, serviceOptions v3.Kubern
 		// cri-dockerd must be enabled if the cluster version is 1.24 and higher
 		if parsedRangeAtLeast124(parsedVersion) {
 			CommandArgs["container-runtime-endpoint"] = "unix:///var/run/cri-dockerd.sock"
-			Binds = []string{fmt.Sprintf("%s:/var/lib/cri-dockerd:z", path.Join(host.PrefixPath, "/var/lib/cri-dockerd"))}
+			Binds = []string{fmt.Sprintf("%s:/var/lib/cri-dockerd:z", path.Join(host.PrefixPath, "/var/lib/cri-dockerd")),
+				fmt.Sprintf("%s:%s", path.Join(host.PrefixPath, KubeletDockerConfigPath), "/.docker/config.json")}
```

Comment:
hi @kinarashah , Although I approved the PR, I just had this question and want to get confirmation from you:

The binds are added when the cluster version is at least 1.24, so they are not applied when cri-dockerd is enabled on cluster <= 1.23, right?  Why do we do it this way, is it because the problem does not exist for the older version of cri-dockerd used on cluster < 1.23?

Reply:
nope you're right, this should be outside the v1.24 block but under `IsCRIDockerdEnabled` check..

Reply:
@jiaqiluo On second thought, I’m going to hold off on not adding change for 1.23 or older clusters…we’ll have to hardcode checks for latest patch versions here in order to avoid cluster auto upgrades due to change in kubelet process. 

We do have a workaround if users want to enable cri-dockerd on <=1.23 clusters and use private registry, so I think we should be good there. 

The workaround is to add the same bind mount in cluster.yml under services/kubelet/extra_binds.

Reply:
So we can merge this PR if the change looks good for 1.24 and if needed, will put up a new PR for other k8s versions.